ci-automation/sdk_container.sh

```diff
@@ -75,5 +81,19 @@ function _sdk_container_build_impl() {
     docker_image_to_buildcache "${CONTAINER_REGISTRY}/flatcar-sdk-all" "${docker_vernum}"
     docker_image_to_buildcache "${CONTAINER_REGISTRY}/flatcar-sdk-amd64" "${docker_vernum}"
     docker_image_to_buildcache "${CONTAINER_REGISTRY}/flatcar-sdk-arm64" "${docker_vernum}"
+
+    # If the version is not a release, we do not publish on GHCR.
+    [[ "${docker_vernum}" =~ ^[0-9.]+$ ]] || return
+
+    # If the registry password or the registry username is not set, we leave early.
+    [[ -z "${REGISTRY_PASSWORD}" ]] || [[ -z "${REGISTRY_USERNAME}" ]] && return
+
+    # We assert that we are correctly logged into the container registry.
+    echo "${REGISTRY_PASSWORD}" | docker login "${CONTAINER_REGISTRY}" -u "${REGISTRY_USERNAME}" --password-stdin
+
+    # Docker images are pushed in the container registry.
+    for a in all amd64 arm64; do
+      docker push "${CONTAINER_REGISTRY}/flatcar-sdk-${a}":"${docker_vernum}"
```

Comment:
For the future: In case we build something releasable every night, I wouldn't publish it directly here in this job but do the SDK publishing in the release job which is run manually - otherwise we clutter the registry too much.
For now it's ok to do it here, no problem.

Reply:
Or as Gabriel suggested, we use garbage collection to remove the unused images after one week, only keeping those that are used in the version.txt files of the releases under CHANNEL.release.flatcar-linux.net

Reply:
quickly checked, but I don't see a limitation for GH Container Registry? In that case, we could simply push each built SDK, without pruning them.

Reply:
Deleting development SDKs makes sense to give people a chance to mirror the registry efficiently and also helps to reduce the environmental impact ;)

Reply:
I just been through the `release` job - I think that would make more sense to move theses lines directly into the release script.

Reply:
As you like. One argument for doing it here would be that we could skip the tar ball import to create the local Docker image which takes a couple of minutes. Not sure if this needs some other changes in the pipeline and we would still want to keep the possibility to work without a registry, I think?

Reply:
Should we move this to https://github.com/flatcar/scripts/pull/475?

Reply:
Yes, let's do this.